test/renderers/categorization.control.test.ts

```diff
@@ -0,0 +1,191 @@
+import 'jsdom-global/register';
+import * as installCE from 'document-register-element/pony';
+declare var global;
+installCE(global, 'force');
+import {JsonSchema} from '../../src/models/jsonSchema';
+import {Categorization, Category, ControlElement} from '../../src/models/uischema';
+import {CategorizationRenderer, categorizationTester}
+  from '../../src/renderers/additional/categorization-renderer';
+import {DataService } from '../../src/core/data.service';
+import {JsonFormsHolder} from '../../src/core';
+import {Runtime} from '../../src/core/runtime';
+
+test('CategorizationTester', t => {
+  t.is(categorizationTester(undefined, undefined), -1);
+  t.is(categorizationTester(null, undefined), -1);
+  t.is(categorizationTester({type: 'Foo'}, undefined), -1);
+  t.is(categorizationTester({type: 'Categorization'}, undefined), -1);
+  t.is(categorizationTester({type: 'Categorization', elements: null} as Categorization,
+    undefined), -1);
+  t.is(categorizationTester({type: 'Categorization', elements: []} as Categorization,
+    undefined), -1);
+  t.is(categorizationTester(
+    {type: 'Categorization', elements: [{type: 'Foo'}]} as Categorization,
+    undefined), -1);
+  t.is(categorizationTester(
+    {type: 'Categorization', elements: [{type: 'Category'}]} as Categorization,
+    undefined), 1);
+  t.is(categorizationTester(
+    {type: 'Categorization',
+      elements: [{type: 'Categorization', elements: [{type: 'Category'}]}]} as Categorization,
+    undefined), 1);
+  t.is(categorizationTester(
+    {type: 'Categorization',
+      elements: [{type: 'Categorization'}]} as Categorization,
+    undefined), -1);
+  t.is(categorizationTester(
+    {type: 'Categorization',
+      elements: [{type: 'Categorization', elements: null}]} as Categorization,
+    undefined), -1);
+  t.is(categorizationTester(
+    {type: 'Categorization',
+      elements: [{type: 'Categorization', elements: []}]} as Categorization,
+    undefined), -1);
+});
+test('CategorizationRenderer static', t => {
+  const schema = {type: 'object', properties: {name: {type: 'string'}}} as JsonSchema;
+  const renderer: CategorizationRenderer = new CategorizationRenderer();
+  const data = {'name': 'Foo'};
+  renderer.setDataService(new DataService(data));
+  renderer.setDataSchema(schema);
+  renderer.setUiSchema({type: 'Categorization', elements: [
+      {type: 'Categorization', label: 'Bar', elements: [
+        {type: 'Category', label: 'A', elements: [
+          {type: 'Control', scope: {$ref: '#/properties/name'}} as ControlElement]} as Category
+      ]} as Categorization,
+      {type: 'Category', label: 'B', elements: [
+        {type: 'Control', scope: {$ref: '#/properties/name'}} as ControlElement]} as Category
+    ]} as Categorization);
+  const result = renderer.render();
+  t.is(result.className, 'jsf-categorization')
+  t.is(result.childNodes.length, 2);
+  // master tree
+  const master = <HTMLDivElement>result.children[0]; // <-- TODO needed?
```

Comment:
i meant whether we need a div wrapper here or whether we can directly add the ul element